### PR TITLE
Enforce using C++11

### LINKED ISSUE #550: 1.11.2  testsuite error on i386
Hello,

the new release throws a testsuite error on i386:
~~~
cd /dev/shm/VIGRA/libvigraimpex-1.11.2/obj.i386-linux-gnu/test/graph_algorithm && ./run_test_graph_algorithm.sh
Entering test suite GraphAlgorithmTestSuite

Failure in GraphAlgorithmTest::testEdgeWeightComputation()
Assertion failed: Sequences differ at index 5 [4.24264 != 4.24264] (./test/graph_algorithm/test.cxx:543)

1 of 6 tests failed in test suite GraphAlgorithmTestSuite
Leaving test suite GraphAlgorithmTestSuite

make[5]: *** [test/graph_algorithm/CMakeFiles/test_graph_algorithm.dir/build.make:118: test/graph_algorithm/test_graph_algorithm] Error 1
~~~
Full log: https://buildd.debian.org/status/fetch.php?pkg=libvigraimpex&arch=i386&ver=1.11.2%2Bdfsg-1&stamp=1696169408&raw=0
cu Andreas

### PR DESCRIPTION
**Pull request by AdrianBunk**
Support for older standards was dropped in #507

Keeping the default CMAKE_CXX_EXTENSIONS=ON fixes #550

### CODE REVIEW

File: config/VigraDetectThreading.cmake [hunk 1]
```diff
@@ -22,30 +22,24 @@ else()
 
     if (CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.7.0")
         SET(CXX_THREADING_FLAG  "-pthread")
-        SET(CXX_VERSION_FLAG    "-std=c++0x")
     elseif(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
```
Comment (lnw) [inline config/VigraDetectThreading.cmake:25]: The first two blocks could be summarised, by just dropping the first one.

**Comment by hmaarrfk** [comment]
@k-dominik do we want to consider sometihng like this? again, what is our C++ support policy? Do we want to support C++11 still?

**Comment by k-dominik** [comment]
for the particular case we would still be able to support cpp11: https://en.cppreference.com/w/cpp/algorithm/random_shuffle, as `std::shuffle` is already part of that. I'd say we support it until we find it too painful? :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -31,6 +31,9 @@ if(SUPPRESS_3RD_PARTY_WARNINGS)
     set(SUPPRESS_WARNINGS SYSTEM)
 endif()
 
+set(CMAKE_CXX_STANDARD 11)
```
Comment (hmaarrfk) [inline CMakeLists.txt:34]:
```suggestion
set(CMAKE_CXX_STANDARD 11 CACHE)
```

lets let users overwrite this themselves too?

**Pull request merged**